src/sql/src/plan.rs

```diff
@@ -571,17 +573,17 @@ impl QueryWhen {
         }
     }
     /// Returns whether the candidate must be advanced to the upper.
-    pub fn advance_to_upper(&self, uses_tables: bool) -> bool {
+    pub fn advance_to_upper(&self) -> bool {
         match self {
-            QueryWhen::Immediately | QueryWhen::AtLeastTimestamp(_) => !uses_tables,
+            QueryWhen::Immediately | QueryWhen::AtLeastTimestamp(_) => true,
             QueryWhen::AtTimestamp(_) => false,
         }
     }
-    /// Returns whether the candidate must be advanced to the global table timestamp.
-    pub fn advance_to_table_ts(&self, uses_tables: bool) -> bool {
+    /// Returns whether the candidate must be advanced to the global timestamp.
+    pub fn advance_to_global_ts(&self) -> bool {
         match self {
-            QueryWhen::Immediately | QueryWhen::AtLeastTimestamp(_) => uses_tables,
-            QueryWhen::AtTimestamp(_) => false,
+            QueryWhen::Immediately => true,
+            QueryWhen::AtLeastTimestamp(_) | QueryWhen::AtTimestamp(_) => false,
```

Comment:
I just want to highlight that `AtTimestamp` was move to always be false here. The rationale being that if someone uses `AS OF` then they are opting out of consistency guarantees.